### PR TITLE
TF-2199 Email suggestion for unknown email

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
---
#2199 

## Resolved 
---
- Case 1: Suggestion list is empty, but typed text is an email

https://github.com/linagora/tmail-flutter/assets/80142234/db21ee77-7e43-4dfa-bfac-0fba703a0d6b

- Case 2: Suggestion list is not empty, and not include typed email

https://github.com/linagora/tmail-flutter/assets/80142234/8fce2fe7-f172-44dd-9c78-6eaf90b88020

- Android:

https://github.com/linagora/tmail-flutter/assets/80142234/0ed63ffa-942e-44f1-9775-ea8b08b8ca66

### CODE REVIEW

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -940,9 +940,26 @@ class ComposerController extends BaseController {
         .execute(AutoCompletePattern(word: word, accountId: accountId))
         .then((value) => value.fold(
           (failure) => <EmailAddress>[],
-          (success) => success is GetAutoCompleteSuccess
-              ? success.listEmailAddress
-              : <EmailAddress>[]));
+          (success) {
+            if (success is GetAutoCompleteSuccess) {
+              if (success.listEmailAddress.isEmpty == true && GetUtils.isEmail(word) == true) {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:945]:
```suggestion
if (success.listEmailAddress.isEmpty == true && GetUtils.isEmail(word)) {
```

File: lib/features/composer/presentation/composer_controller.dart [hunk 2]
```diff
@@ -940,9 +940,26 @@ class ComposerController extends BaseController {
...
+                final unknownEmailAddress = EmailAddress(word, word);
+                return <EmailAddress>[unknownEmailAddress];
+              }
+              if (GetUtils.isEmail(word) == true) {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:949]:
```suggestion
if (GetUtils.isEmail(word)) {
```

File: lib/features/composer/presentation/composer_controller.dart [hunk 3]
```diff
@@ -952,10 +969,27 @@ class ComposerController extends BaseController {
...
+                  if (!isContainsTypedEmail) {
+                    final unknownEmailAddress = EmailAddress(word, word);
+                    success.listEmailAddress.insert(0, unknownEmailAddress);
+                    return success.listEmailAddress;
+                  }
+                }
+                return success.listEmailAddress;
+              } else {
+                return <EmailAddress>[];
+              }
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:990]: Should be create function handle success to reuse it above

**Comment by chibenwa** [comment]
Thank YOU so MUCH!

I was about to raise this bug again.

Happy to see a fix for this!

**Review by dab246 — APPROVED** [review]

**Pull request merged**